podcasts/New Search/Helpers/SearchAnalyticsHelper.swift

```diff
@@ -0,0 +1,85 @@
+import SwiftUI
+import PocketCastsServer
+
+class SearchAnalyticsHelper: ObservableObject {
+    let source: AnalyticsSource
+
+    init(source: AnalyticsSource) {
+        self.source = source
+    }
+
+    // MARK: - Search
+
+    func trackShown() {
+        Analytics.track(.searchShown, properties: ["source": source])
+    }
+
+    func trackSearchPerformed() {
+        Analytics.track(.searchPerformed, properties: ["source": source])
+    }
+
+    func trackFailed() {
+        Analytics.track(.searchFailed, properties: ["source": source])
+    }
+
+    func trackResultTapped(_ searchResult: Any) {
+        var type = "unknown"
+        var uuid = ""
+        if let folderOrPodcast = searchResult as? PodcastFolderSearchResult {
+            type = folderOrPodcast.type
+            uuid = folderOrPodcast.uuid
+        } else if let episode = searchResult as? EpisodeSearchResult {
+            type = "episode"
+            uuid = episode.uuid
+        }
+        Analytics.track(.searchResultTapped, properties: ["source": source, "uuid": uuid, "result_type": type])
+    }
```

Comment:
What do you think about specifying a type here?

We could do something like this:

```
protocol AnalyticsSearchResultItem: AnalyticsDescribable {
    var uuid: String { get }
}

extension EpisodeSearchResult: AnalyticsSearchResultItem {
    var analyticsDescription: String {
        "episode"
    }
}

extension PodcastFolderSearchResult: AnalyticsSearchResultItem {
    var analyticsDescription: String {
        if isFolder == true {
            return "folder"
        } else if isLocal == true {
            return "podcast_local_result"
        } else {
            return "podcast_remote_result"
        }
    }
}
```

Which would change this to: 
```suggestion
    func trackResultTapped(_ searchResult: AnalyticsSearchResultItem) {
        Analytics.track(.searchResultTapped, properties: ["source": source, "uuid": searchResult.uuid, "result_type": searchResult])
    }
```